lib/module.js

```diff
@@ -434,7 +435,34 @@ Module._load = function(request, parent, isMain) {
     debug('Module._load REQUEST %s parent: %s', request, parent.id);
   }
 
-  var filename = Module._resolveFilename(request, parent, isMain);
+  var filename = null;
+
+  if (isMain) {
+    let err;
+    try {
+      filename = Module._resolveFilename(request, parent, isMain);
+    } catch (e) {
+      // try to keep stack
+      e.stack;
+      err = e;
+    }
+    if (filename === null || /\.mjs$/.test(filename)) {
+      try {
+        ESM.loader.import(request).catch((e) => {
+          console.error(e);
+          process.exit(1);
+        });
+        return;
```

Comment:
I noticed when loading a file with an error via `node foo.mjs` I get
```
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): SyntaxError: The requested module does not provide an export named 'x'
(node:87429) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
There seems to be no way to catch the rejection.